File: week07/LabeledExprPy/Calc.py
import sys
from antlr4 import *
from LabeledExprLexer import LabeledExprLexer
from LabeledExprParser import LabeledExprParser
from EvalVisitorImpl import EvalVisitorImpl

def main():
    if len(sys.argv) <= 1:
        return -1
    input_stream = FileStream(sys.argv[1])
    lexer = LabeledExprLexer(input_stream)
    tokens = CommonTokenStream(lexer)
    parser = LabeledExprParser(tokens)
    tree = parser.prog()
    eval = EvalVisitorImpl()
    eval.visitProg(tree)

if __name__ == '__main__':
    main()

File: TF/aux/Calc.cpp
#include <cstdio>
#include <fstream>
#include <string>

#include "EvalVisitorImpl.h"
#include "antlr4-runtime.h"
#include "LabeledExprLexer.h"
#include "LabeledExprParser.h"

/*import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.FileInputStream;
import java.io.InputStream;
*/

int main(int argc, char** argv) {
    std::ifstream inputFile;
    if (argc > 1) {
        inputFile.open(argv[1]);
    }
    antlr4::ANTLRInputStream input(inputFile.is_open() ? inputFile : std::cin);
    LabeledExprLexer lexer(&input);
    antlr4::CommonTokenStream tokens(&lexer);
    LabeledExprParser parser(&tokens);
    auto tree = parser.prog();
    EvalVisitorImpl eval;
    eval.visitProg(tree);
    return 0;
}

//clang++ -std=c++17  -I $ANTLR4I -L $ANTLR4L -lantlr4-runtime  *.cpp
//g++ *.cpp -I $ANTLR4I -L $ANTLR4L -lantlr4-runtime